src/stactools/sentinel1/grd/product_metadata.py

```diff
@@ -52,7 +60,7 @@ def _get_geometries():
             # Convert to values
             footprint_value = [
                 float(x)
-                for x in footprint_text[0].text.replace(" ", ",").split(",")
+                for x in str(footprint_text[0].text).replace(" ", ",").split(",")
```

Comment:
Given the preceding suggestion, the line above should become the following:

```suggestion
                for x in footprint_text.replace(" ", ",").split(",")
```

At this point, the type of `footprint_text` is automatically narrowed to `str` (instead of `str | None`) since it was checked against `None` earlier, and raised an error in that case.